SillyBusinessInc/SillyBusinessGame
# Fix/#97 camera fix

## Purpose of this PR:
Fixing some issues designers noticed with the camera. 
Removed pan tilt in favor of Hard Looks at Target rotation controls.
Refactor of the player prefab to take the camera out and make it it's own prefab. We dont want the cinemachine brain and camera to inherit the positions and transforms of the player, therefore i had to break it out into it's own object and prefab.
I've also added a Smoothed Target object to the player, this is the target we want to track with the camera. This Smoothed Target needs a reference to the player's camera target object. It also allows you to change to smoothing speed (default: 5)
This ticket also adds Camera Deocclusion, the camera will get closer to the player if its view were to be obstructed. This is set to ignore the following Unity Layers: Player, Enemies.

## How to test:
Open the FixFollowEnemyAttack scene, as this one should be properly setup with the new camera and player prefab and references. 
walk around, move around, get some speed and see if the camera follows okay. 
now go stand in front of a wall , and aim the camera at a wall so the character and wall are in view at the same time. quickly change directions by tapping A and D, or W and S. Check if there is still a noticable strobing effect in the background (the wall) due to the sudden rotation of the player (There is some room for the designers to fine tune this by tweaking the smoothing speed).
To check the deocclusion, get close to a wall and turn your camera so the wall will be between the camera and the playerObject. Verify the camera instead stays in front of said wall.
Test that enemies dont cause this same effect, and that the player's own model doesnt trigger this either (walking towards the camera can cause this effect if layers arent properly set up)

### links: 
closes #97
